### ISSUE TITLE
Entity Linking : Empty Kb being generated while generating KB with wikidata

### ISSUE THREAD

**Issue by akhilnegi** [explosion/spaCy#4870]
## How to reproduce the behavior
`python wikidata_pretrain_kb.py 'latest-all.json.bz2' 'enwiki-latest-pages-articles-multistream.xml.bz2' './output' 'en_core_web_lg' `

## Your Environment
spaCy version    2.2.1
Platform         Windows-10-10.0.18362-SP0
Python version   3.6.9

## Expected behaviour
This should generate a knowledge base using Wikidata.

## What is happening
The step ran for 18 hours and generated an ~empty KB file, below is how the output directory looks:

│   entity_alias.csv	206,414 KB
│   entity_defs.csv	172,606 KB
│   entity_descriptions.csv 153,340 KB
│   entity_freq.csv	1 KB
│   gold_entities.jsonl	0 KB
│   kb			1 KB
│   prior_prob.csv	1 KB
└───nlp_kb
    │   meta.json	2 KB
    │   tokenizer	68 KB
    ├───ner
    │       cfg		1 KB
    │       model	4147 KB
    │       moves	2 KB
    ├───parser
    │       cfg		1KB
    │       model	4253 KB
    │       moves	2 KB
    ├───tagger
    │       cfg		1 KB
    │       model	3856 KB
    │       tag_map	2 KB
    └───vocab
            key2row	9235 KB
            lexemes.bin	125648 KB
            lookups.bin	1658 KB
            strings.json 23573 KB
            vectors	802537 KB

**Comment by svlandeg** [comment]
It looks like `prior_prob.csv` was not properly created from Wikipedia, and thus the required information was not available to create the entities for the DB. But other parts like aliases (also parsed from Wikipedia) did go through correctly.

I'm guessing this happened because maybe you still had a few (empty) `.csv` files in that output directory, and then the script skipped those steps in order not to duplicate any previous efforts.

You could use this to your advantage though: remove `entity_freq.csv`, `gold_entities.jsonl`, `kb`, `prior_prob.csv` and the whole `nlp_kb` folder.
You can now run the script again and it should create the missing files.

The other option (perhaps easier for debugging) is to start fresh with an entirely clean output folder, and run the command again, but specify the 3 limit parameters as explained [here](https://github.com/explosion/spaCy/tree/master/bin/wiki_entity_linking#step-1-create-a-knowledge-base-kb-and-training-data). For instance, try setting them like so: `-lt 20000 -lp 2000 -lw 3000 -f 1`. This will not create the full KB, but will perform a quick test run that finishes much faster so we can see what's going on more quickly. Paste the full output log of the script here in case there's still issues.

Finally, I'd like to recommend updating spaCy or even building the current `master` branch. This EL functionality is still in heavy development and is still changing quite a bit, and it may be that there was some spurious bug in 2.2.1 that got fixed in the meantime ... In fact, I would recommend building the code from PR https://github.com/explosion/spaCy/pull/4811, as this is less memory intensive.

Let me know what you find!

**Comment by akhilnegi** [comment]
Hi @svlandeg 

I was able to fix the issue. The issue was due to corruption in 'enwiki-latest-pages-articles-multistream.xml.bz2' file. Once I downloaded it again, I was able to get  below entities and aliases with the default settings:

`2020-01-07 09:28:59,692 - INFO - __main__ - kb entities: 1119062`
`2020-01-07 09:28:59,693 - INFO - __main__ - kb aliases: 1627699`

My bad, I should have checked this before raising the issue.
Thanks for the quick reply though. And yes, the params for quick testing helped a great deal.
